dart/pubspec.yaml

```diff
@@ -13,6 +13,7 @@ dependencies:
   meta: ^1.0.0
   stack_trace: ^1.0.0
   uuid: ^2.0.0
+  collection: ^1.14.13
```

Comment:
We could drop this dependency if we don't have a scope stack?
Would this be a reason to just go with a plain `Scope` field in the `Hub` instead?
The code will be MUCH simpler in that case.

Reply:
here, this package is only used for lists and maps "deep comparaisons" in `==` operator overrides

Reply:
Maybe we should consider dropping the package and the overrides if there's an alternative.
For context: https://develop.sentry.dev/sdk/philosophy/#dependencies-cost

Reply:
✅ moved to the devDependencies